### PR TITLE
Feature: Allow multiple parentCategories

### PR DESCRIPTION
**Pull request by Bilelkihal**
Require: https://github.com/ontoportal-lirmm/ontologies_linked_data/pull/162
### Changes
- Update categories selector in the admin page to allow the selection of multiple parent categories (https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/772/commits/b55e1b30d45cb335b60d93a6acfe21336b330ce6)
- Handle multiple categories in categories page (display them multiple times under each parent) (https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/772/commits/81b2aee856b23c9077a71d45ac7969a233ffb428)

### Screenshots:
image

image

### CODE REVIEW

File: app/controllers/taxonomy_controller.rb [hunk 1]
```diff
@@ -33,13 +32,13 @@ def nest_categories_children(categories)
...
-      if category.parentCategory
-        parent = category_index[category.parentCategory]
+      category[:parentCategory].each do |parent_id|
+        parent = category_index[parent_id]
         parent[:children] ||= []
         parent[:children] << category
       end
     end
-    categories.reject! { |category| category.parentCategory }
+    categories.reject! { |category| category[:parentCategory].any? }
```
Comment (syphax-bouazzouni) [inline app/controllers/taxonomy_controller.rb:41]: is it not better like this `categories.select! { |category| category[:parentCategory].blank? }`, but not sure to test.

Comment (Bilelkihal) [reply app/controllers/taxonomy_controller.rb:41]: It's the inverse actually (delete the categories that have at least one parent)
and I didn't want to do `!category[:parentCategory].blank?`

Comment (syphax-bouazzouni) [reply app/controllers/taxonomy_controller.rb:41]: I'm not sure that `any?` does work the way you would expect, but maybe I'm wrong

Comment (syphax-bouazzouni) [reply app/controllers/taxonomy_controller.rb:41]: but let's assume it is for now

**Review by syphax-bouazzouni — APPROVED** [review]

File: app/controllers/taxonomy_controller.rb (hunk 1 above)
Comment (syphax-bouazzouni) [reply app/controllers/taxonomy_controller.rb:41]: ok. maybe you just need to add `&` to safe check nil
image

Comment (Bilelkihal) [reply app/controllers/taxonomy_controller.rb]: Done

**Comment by jonquet** [comment]
Sounds good.

**Pull request merged**